core/shared/src/main/scala/fs2/Stream.scala

```diff
@@ -2062,8 +2062,18 @@ final class Stream[+F[_], +O] private[fs2] (private[fs2] val underlying: Pull[F,
     */
   def parEvalMapUnbounded[F2[x] >: F[x], O2](f: O => F2[O2])(implicit
       F: Concurrent[F2]
-  ): Stream[F2, O2] =
-    parEvalMap[F2, O2](Int.MaxValue)(f)
+  ): Stream[F2, O2] = {
+
+    def init(ch: Channel[F2, F2[Either[Throwable, O2]]], release: F2[Unit]) =
+      Deferred[F2, Either[Throwable, O2]].flatTap { value =>
+        ch.send(release *> value.get)
+      }
+
+    def send(v: Deferred[F2, Either[Throwable, O2]]) =
+      (el: Either[Throwable, O2]) => v.complete(el).void
```

Comment:
Rather than a function that builds a `lambda`, could we make this two parameter lists?
```suggestion
    def send(v: Deferred[F2, Either[Throwable, O2]])(el: Either[Throwable, O2]): F[Unit] =
      v.complete(el).void
```

Reply:
I think this change will cause an unnecessary indirection, since ultimately we need a lambda.

```scala
def send(v: Deferred[F2, Either[Throwable, O2]])(el: Either[Throwable, O2]): F[Unit]
```

will desugar to
```scala
def send(v: Deferred[F2, Either[Throwable, O2]], el: Either[Throwable, O2]): F[Unit]
```
So then we still need to allocate a lambda that then delegates to `send`.

Whereas the current version allocates a lambda that directly invokes `v.complete(el).void`, no indirection.